ScrapPackedTests/TestCliApp.cs

```diff
@@ -289,114 +246,304 @@ public void TestRunExtract()
+            //var fsFile = new FileStream(@"TestResults\TestBackup\examplefile1.txt", FileMode.Open);
+            //try
+            //{
+            //    CheckRunCompareFile(new[] {"extract", "--packedFile", @"TestResults\TestBackup\packedFile.packed",
+            //    "--destinationPath", @"TestResults\TestBackup\examplefile1.txt",
+            //    "--packedPath", @"file.txt" },
+            //    @"TestResults\TestBackup\examplefile1.txt",
+            //    @"TestData\examplefile3.txt",
+            //    "Restore backup of file after failng extraction", 1
+            //    );
+            //}
+            //finally
+            //{
+            //    fsFile.Close();
+            //}
+
+            CheckRunFileExists(new[] {"extract", "--packedFile", @"TestResults\TestBackup\packedFile.packed",
+                "--destinationPath", @"TestResults\TestBackup\examplefile1.txt",
+                "--packedPath", @"file.txt" },
+                @"TestResults\TestBackup\examplefile1.txt.bak",
+                "Backup of alredy extracted file after extraction"
+            );
+
+            // todo: keepBackup & overwriteOldBackup
+            // Todo: implement check later
+            /* // add file keep backup
+            // (create dummy packedFile.packed.bak before call, expect to be unchanged)
+            CheckRun(new[] { "add", "--packedFile", @"TestResults\TestAdd\packedFile.packed", "--sourcePath", "'examplefile1.txt'", "--packedPath", "'folder/file.txt'", "--keepBackup" }, "", "");
+            // add file keep backup but overwrite old backup
+            CheckRun(new[] { "add", "--packedFile", @"TestResults\TestAdd\packedFile.packed", "--sourcePath", "'examplefile3.txt'", "--packedPath", "'folder/file.txt'", "--keepBackup", "--overwriteOldBackup" }, "", "");
+            // add file keep do not keep backup and old backup
+            // (create dummy packedFile.packed.bak before call, expect to be unchanged)
+            CheckRun(new[] { "add", "--packedFile", @"TestResults\TestAdd\packedFile.packed", "--sourcePath", "'examplefile2.png'", "--packedPath", "'folder/file.png'", "--overwriteOldBackup" }, "", "");
+            */
         }
 
-        private void CheckRunCompareFile(string[] p_Args, string p_ExpectedFilePath, string p_ActualFilePath, string p_Message = "")
+        private void CheckRunCompareFile(string[] p_Args, string p_ExpectedFilePath, string p_ActualFilePath, string p_Message = "", int p_ReturnCode = 0)
         {
             var cliApp = new CliApp();
             var returnValue = cliApp.Run(p_Args);
 
-            Assert.AreEqual(0, returnValue, p_Message+": wrong return value");
+            Assert.AreEqual(p_ReturnCode, returnValue, p_Message+": wrong return value");
             AssertFilesEqual(p_ExpectedFilePath, p_ActualFilePath, p_Message + ": files differ");
         }
 
+        // todo: remove this method?
```

Comment:
why? it is used 15 times …
oh you mean replace all usages with one of the other more sophisticated check methods?

Reply:
Yes